Map View, Quick Settings -> "Show direction"

Find a better wording for that function as "Show direction" is potentially misleading. See here: https://github.com/cgeo/cgeo/commit/b424f1b21f598887903d2877c564750a0e890701#commitcomment-46373202 and the following comments.

We have agreed to call it: 

"Show routing line"

Although it is a rather trivial change in Strings.xml I am opening an issue here such that I can reference it in the process of creating a pull request. I will use this as an excercise to learn / refresh how the process is working.

For completeness, here is the FATAL ABORT for the Grant, do nothin scenario
`2021-01-30 10:25:05.079 12279-12279/cgeo.geocaching D/AndroidRuntime: Shutting down VM
2021-01-30 10:25:05.080 12279-12279/cgeo.geocaching E/AndroidRuntime: FATAL EXCEPTION: main
    Process: cgeo.geocaching, PID: 12279
    java.lang.RuntimeException: Failure delivering result ResultInfo{who=null, request=59372, result=-1, data=Intent { dat=content://com.android.providers.downloads.documents/tree/downloads flg=0xc3 }} to activity {cgeo.geocaching/cgeo.geocaching.MainActivity}: java.lang.IllegalArgumentException: mimeType can't be null
        at android.app.ActivityThread.deliverResults(ActivityThread.java:4845)
        at android.app.ActivityThread.handleSendResult(ActivityThread.java:4886)
        at android.app.servertransaction.ActivityResultItem.execute(ActivityResultItem.java:51)
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2016)
        at android.os.Handler.dispatchMessage(Handler.java:107)
        at android.os.Looper.loop(Looper.java:214)
        at android.app.ActivityThread.main(ActivityThread.java:7356)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:930)
     Caused by: java.lang.IllegalArgumentException: mimeType can't be null
        at android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:170)
        at android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:140)
        at android.content.ContentProviderProxy.call(ContentProviderNative.java:658)
        at android.content.ContentResolver.call(ContentResolver.java:2042)
        at android.provider.DocumentsContract.createDocument(DocumentsContract.java:1327)
        at cgeo.geocaching.storage.DocumentContentAccessor.createInternal(DocumentContentAccessor.java:111)
        at cgeo.geocaching.storage.DocumentContentAccessor.create(DocumentContentAccessor.java:98)
        at cgeo.geocaching.storage.ContentStorage.create(ContentStorage.java:169)
        at cgeo.geocaching.storage.ContentStorage.tryTestReadWriteToFolder(ContentStorage.java:426)
        at cgeo.geocaching.storage.ContentStorage.ensureFolder(ContentStorage.java:133)
        at cgeo.geocaching.storage.ContentStorageActivityHelper.handleResultPersistableFolderSelection(ContentStorageActivityHelper.java:338)
        at cgeo.geocaching.storage.ContentStorageActivityHelper.onActivityResult(ContentStorageActivityHelper.java:210)
        at cgeo.geocaching.activity.AbstractActivity.onActivityResult(AbstractActivity.java:300)
        at cgeo.geocaching.MainActivity.onActivityResult(MainActivity.java:571)
        at android.app.Activity.dispatchActivityResult(Activity.java:8110)
        at android.app.ActivityThread.deliverResults(ActivityThread.java:4838)
        at android.app.ActivityThread.handleSendResult(ActivityThread.java:4886) 
        at android.app.servertransaction.ActivityResultItem.execute(ActivityResultItem.java:51) 
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135) 
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2016) 
        at android.os.Handler.dispatchMessage(Handler.java:107) 
        at android.os.Looper.loop(Looper.java:214) 
        at android.app.ActivityThread.main(ActivityThread.java:7356) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:930) 
`

--- a/main/res/values/strings.xml
+++ b/main/res/values/strings.xml
@@ -1835,10 +1835,6 @@
     <string name="do_not_ask_me_again">Do not ask me again</string>
     <string name="geokrety_parsing_failed">Failed to parse GeoKrety response.</string>
 
-    <string name="settings_map_direction">Show direction</string>
-    <string name="settings_summary_map_direction">Show a direction line to the navigation target</string>
-    <string name="map_direction">Show direction</string>
-
     <string name="command_move_caches_progress">Moving caches to list %1s</string>
     <plurals name="command_move_caches_result">
         <item quantity="zero">Moved %d caches</item>

--- a/main/res/xml/preferences.xml
+++ b/main/res/xml/preferences.xml
@@ -624,11 +624,6 @@
         <PreferenceCategory android:title="@string/settings_title_map_content"
             android:layout="@layout/preference_category" >
             <CheckBoxPreference
-                android:defaultValue="true"
-                android:key="@string/pref_map_direction"
-                android:summary="@string/settings_summary_map_direction"
-                android:title="@string/settings_map_direction" />
-            <CheckBoxPreference
                 android:defaultValue="false"
                 android:key="@string/pref_maptrail"
                 android:summary="@string/init_summary_maptrail"

--- a/main/src/cgeo/geocaching/maps/routing/Routing.java
+++ b/main/src/cgeo/geocaching/maps/routing/Routing.java
@@ -210,7 +210,7 @@ public final class Routing {
         // wanted to compare (gpx.substr(0,5) != "<?xml")
         // but this did not work, most likely because gpx is a Unicode string?
 
-        if (gpx.length() < 200) {
+        if (!gpx.startsWith("<?xml")) {
             Log.w("brouter returned an error message: " + gpx);
             return null;
         }

--- a/main/src/cgeo/geocaching/maps/routing/Routing.java
+++ b/main/src/cgeo/geocaching/maps/routing/Routing.java
@@ -237,7 +237,8 @@ public final class Routing {
             return result.toArray(new Geopoint[result.size()]);
 
         } catch (final SAXException e) {
-            Log.w("cannot parse brouter output of length " + gpx.length(), e);
+            //Log.w("cannot parse brouter output of length " + gpx.length(), e);
+            Log.w("cannot parse brouter output of length " + gpx.length() + ", gpx=" + gpx, e);
         }
         return null;
     }

--- a/main/src/cgeo/geocaching/maps/routing/Routing.java
+++ b/main/src/cgeo/geocaching/maps/routing/Routing.java
@@ -237,7 +237,6 @@ public final class Routing {
             return result.toArray(new Geopoint[result.size()]);
 
         } catch (final SAXException e) {
-            //Log.w("cannot parse brouter output of length " + gpx.length(), e);
             Log.w("cannot parse brouter output of length " + gpx.length() + ", gpx=" + gpx, e);
         }
         return null;

--- a/main/src/cgeo/geocaching/maps/routing/Routing.java
+++ b/main/src/cgeo/geocaching/maps/routing/Routing.java
@@ -207,8 +207,13 @@ public final class Routing {
             return null;
         }
 
-        // TODO: need to check gpx for error message instead of valid XML
-        // W/cgeo: [LayerManager] cannot parse brouter output of length 29, gpx=datafile E5_N45.rd5 not found
+        // wanted to compare (gpx.substr(0,5) != "<?xml")
+        // but this did not work, most likely because gpx is a Unicode string?
+
+        if (gpx.length() < 200) {
+            Log.w("brouter returned an error message: " + gpx);
+            return null;
+        }
 
         return parseGpxTrack(gpx, dest);
     }

--- a/main/src/cgeo/geocaching/maps/routing/Routing.java
+++ b/main/src/cgeo/geocaching/maps/routing/Routing.java
@@ -206,10 +206,7 @@ public final class Routing {
             Log.i("brouter returned no data");
             return null;
         }
-
-        // wanted to compare (gpx.substr(0,5) != "<?xml")
-        // but this did not work, most likely because gpx is a Unicode string?
-
+        
         if (!gpx.startsWith("<?xml")) {
             Log.w("brouter returned an error message: " + gpx);
             return null;

--- a/main/src/cgeo/geocaching/maps/routing/Routing.java
+++ b/main/src/cgeo/geocaching/maps/routing/Routing.java
@@ -207,6 +207,9 @@ public final class Routing {
             return null;
         }
 
+        // TODO: need to check gpx for error message instead of valid XML
+        // W/cgeo: [LayerManager] cannot parse brouter output of length 29, gpx=datafile E5_N45.rd5 not found
+
         return parseGpxTrack(gpx, dest);
     }
 

--- a/main/src/cgeo/geocaching/settings/Settings.java
+++ b/main/src/cgeo/geocaching/settings/Settings.java
@@ -852,11 +852,7 @@ public class Settings {
      * whether to show a direction line on the map
      */
     public static boolean isMapDirection() {
-        return getBoolean(R.string.pref_map_direction, true);
-    }
-
-    public static void setMapDirection(final boolean showDirection) {
-        putBoolean(R.string.pref_map_direction, showDirection);
+        return Settings.getRoutingMode() != RoutingMode.OFF;
     }
 
     /**

